cleanlab/count.py

```diff
@@ -48,12 +47,12 @@
 )
 
 
-def num_label_errors(
+def num_label_issues(
```

Comment:
Is this function really necessary? 
Can't a user just do `error_inds = find_label_issues(..., return_indices_ranked_by=...)` and then: `len(error_inds)`?

What is user supposed to conclude when `len(error_inds)` does not match output of this function?  This function supports much fewer options than `find_label_issues`, also unclear why that is.

Reply:
[Quality of life improvement] I also advocate `find_label_issues()` should automatically print out the number of label issues during its execution.

Reply:
`find_label_issues()` now prints out number of label issues when `verbose=1`.

`sum(find_label_issues(...))` usually matches `num_label_issues`, but not always. For example, it ignores the `MIN_NUM_PER_CLASS = 1` constraints required when filtering, but most importantly!! its computed **more accurately** than `find_label_issues` -- the reasons is because its computed just based on the trace, ignoring all off-diagonals (which is how find_label_issues works). By simply summing over the diagonals (which are the most accurately computed elements in the joint), you can get a more accurate estimate of the label errors.

This will become the canonical way to find errors simply by combining a ranking/scoring function from `rank.py` with `num_label_issues()`. I'll update the docstring in `num_label_issues()` to reflect this information.

Reply:
docstring now reads

```
Estimates the number of label issues in `labels`.
    
    This method is **more accurate** than `sum(find_label_issues())` because its computed using only
    the Trace(joint), ignoring all off-diagonals (used by `find_label_issues` and harder to
    estimate). Here we sum over only diagonal elements in the joint (which have more data
    are more constrained, and therefore easier to compute).
    
    tl;dr - Use this method to get the most accurate estimate of number of label issues when you
    don't need the indices of the label issues. This is the canonical way to find errors
    simply by combining a ranking/scoring function from `rank.py` with `num_label_issues()`.

Reply:
This part of the docstring may make user feel as if they're missing something:
"This is the canonical way to find errors
simply by combining a ranking/scoring function from `rank.py` with `num_label_issues()`"

Where is an example of this "canonical way" in action?

Reply:
maybe it's clearer as: 
"Another way to find label issues is to simply let `N = num_label_issues()` and then report the top `N`-ranked results based on label quality score from `rank.py`. The advantages/disadvantages of this alternative vs `find_label_issues()` are ..."

But even then it seems confusing to present two ways of doing the same thing, especially when one way has the canonical name: `find_label_issues()`

Reply:
Yeah, I agree with Jonas that this is potentially confusing. I think the default should be that `num_label_issues() = sum(find_label_issues(...))`. A user might check the number of label issues in a dataset to see if it's high enough to be worth investigating further, and then run `find_label_issues`, only to find that the number of 'promised' issues is fewer than expected.

Perhaps, `num_label_issues` can have a toggle argument `use_trace_joint = False`? That way, experienced users can still get the most accurate count if it's important to them, but the average user won't see a mismatch between `num_label_issues()` and `sum(find_label_issues(...))`.

Reply:
Leaving as is for the following reasons:
* default behavior should be the most accurate, which it currently is.
* default behavior should try to be the fastest approach, which it currently is
* `cleanlab` supports multiple approaches to things throughout (e.g. rank_methods, filter_methods, etc). We can also support multiple approaches to find the threshold cut-off for number of label errors in the dataset.

Reply:
There are two possible workflows:
1. 
```
from cleanlab import filter
label_issues_bool = find_label_issues()
# Do whatever
```

2. 
```
from cleanlab import rank, count
label_quality_scores = rank.get_normalized_margin_for_each_label(labels, probs)
num_errors = count.num_label_issues()
# Label issues are the sorted top num_errors lowest scores